pkg/querier/queryrange/index_stats_cache_test.go

```diff
@@ -79,7 +82,7 @@ func TestIndexStatsCache(t *testing.T) {
 	// should reuse part of the previous request and issue a new request for the remaining time till end.
 	// The new start time is 15m (i.e. 25%) in the future with regard to the previous request time span.
 	*calls = 0
-	req := statsReq.WithStartEnd(statsReq.GetStart().Add(15*time.Minute), statsReq.GetEnd().Add(15*time.Minute))
+	req := statsReq.WithStartEnd(statsReq.GetStart().Add(15*time.Minute), statsReq.GetEnd().Add(15*time.Minute)).(queryrangebase.Request)
```

Comment:
this type conversion necessary?

Reply:
No longer necessary since separated `queryrangebase.Request/Response` from `resultscache.Request/Response`. Removed.